api/src/repositories/attachment-repository.ts

```diff
@@ -113,7 +113,155 @@ export class AttachmentRepository extends BaseRepository {
     // TODO
   }
 
+  /**
+   * SQL query to delete security for Project Attachment
+   *
+   * @param {number} securityId
+   * @param {number} attachmentId
+   * @return {*}  {Promise<{ project_attachment_persecution_id: number }>}
+   * @memberof AttachmentRepository
+   */
+  async removeSecurityFromProjectAttachment(
+    securityId: number,
+    attachmentId: number
+  ): Promise<{ project_attachment_persecution_id: number }> {
+    const sqlStatement = SQL`
+      DELETE FROM
+        project_attachment_persecution
+      WHERE
+        project_attachment_id = ${attachmentId}
+      AND
+        project_attachment_persecution_id =  ${securityId}
+      RETURNING project_attachment_persecution_id
+      ;
+      `;
+
+    const response = await this.connection.sql(sqlStatement);
+
+    const result = (response && response.rows && response.rows[0]) || null;
+
+    if (!result) {
+      throw new ApiExecuteSQLError('Failed to get Delete Project Security Attachment', [
+        'AttachmentRepository->removeSecurityFromProjectAttachment',
+        'row[0] was null or undefined, expected row[0] != null'
+      ]);
+    }
+
+    return result;
+  }
+
+  /**
+   * SQL query to delete security for Survey Attachment
+   *
+   * @param {number} securityId
+   * @param {number} attachmentId
+   * @return {*}  {Promise<{ survey_attachment_persecution_id: number }>}
+   * @memberof AttachmentRepository
+   */
+  async removeSecurityFromSurveyAttachment(
+    securityId: number,
+    attachmentId: number
+  ): Promise<{ survey_attachment_persecution_id: number }> {
+    const sqlStatement = SQL`
+      DELETE FROM
+        survey_attachment_persecution
+      WHERE
+        survey_attachment_id = ${attachmentId}
+      AND
+        survey_attachment_persecution_id =  ${securityId}
+      RETURNING survey_attachment_persecution_id
+      ;
+      `;
+
+    const response = await this.connection.sql(sqlStatement);
+
+    const result = (response && response.rows && response.rows[0]) || null;
+
+    if (!result) {
+      throw new ApiExecuteSQLError('Failed to get Delete Survey Security Attachment', [
```

Comment:
nit: maybe reword to something like `Failed to delete survey attachment security`

Reply:
and below